package/MDAnalysis/analysis/bat.py

```diff
@@ -0,0 +1,503 @@
+            # Find a0, which is a new atom connected to the selected atom
+            a0_list = _sort_atoms_by_mass(a for a in a1.bonded_atoms \
+                if (a in atoms) and (a not in selected_atoms))
+            for a0 in a0_list:
+                # Find a2, which is connected to a1, is not a terminal atom,
+                # and has been selected
+                a2_list = _sort_atoms_by_mass(a for a in a1.bonded_atoms \
+                    if (a!=a0) and len(a.bonded_atoms)>1 and \
+                        (a in atoms) and (a in selected_atoms))
+                for a2 in a2_list:
+                    # Find a3, which is
+                    # connected to a2, has been selected, and is not a1
+                    a3_list = _sort_atoms_by_mass(a for a in a2.bonded_atoms \
+                        if (a!=a1) and \
+                            (a in atoms) and (a in selected_atoms))
+                    for a3 in a3_list:
+                        # Add the torsion to the list of torsions
+                        torsions.append(mda.AtomGroup([a0, a1, a2, a3]))
+                        # Add the new atom to selected_atoms
+                        # which extends the loop
+                        selected_atoms.append(a0)
+                        torsionAdded = True
+                        break # out of the a3 loop
+                    break # out of the a2 loop
+        if torsionAdded is False:
+            print('Selected atoms:')
+            print([a.index+1 for a in selected_atoms])
+            print('Torsions found:')
+            print([list(t.indices+1) for t in torsions])
+            raise ValueError('Additional torsions not found.')
+    return torsions
+
+
+class BAT(AnalysisBase):
+    """Calculate BAT coordinates for the specified AtomGroup.
+
+    BAT coordinates will be computed for the group of atoms and all frames
+    in the trajectory belonging to `ag'.`
+
+    """
+    @due.dcite(Doi("10.1002/jcc.26036"),
+        description="Bond-Angle-Torsions Coordinate Transformation",
+        path="MDAnalysis.analysis.bat.BAT")
+    def __init__(self, ag, initial_atom=None, bat_nc=None, **kwargs):
+        r"""Parameters
+        ----------
+        ag : AtomGroup or Universe
+            Group of atoms for which the BAT coordinates are calculated.
+            ag must have a bonds attribute. If not available,
+            bonds may be guessed using
+            :meth:`AtomGroup.guess_bonds <MDAnalysis.core.groups.AtomGroup.guess_bonds>`.
+            ag must only include one molecule.
+            If a trajectory is associated with the atoms then the computation
+            iterates over the trajectory.
+        initial_atom : Atom
+            The atom whose Cartesian coordinates define the translation
+            of the molecule. If not specified, the heaviest terminal atom
+            will be selected.
+        bat_nc : str
+            File name of a netCDF4 file containing a saved bat attribute.
```

Comment:
`filename`; you need to explain somewhere that you can load the data from a netcdf file instead of using `run()`.